### ISSUE TITLE
The application hangs for seconds with a slow internet connection

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/mozilla-vpn-client#2773]
The issue is reproducible only on windows and it's related to ICMP packages.
In particular, `IcmpCloseHandle` takes seconds to complete the operation.
Using clumsy to slow down my connection, this method takes 7~ seconds to run.

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-1756)

**Comment by data-sync-user** [comment]
➤ Andrea Marchesini commented:

The QA team managed to reproduce this issue (Bianca is about to upload a video + logs). See the extacted log:

```
[07.04.2022 13:38:53.702] Debug: (networking - ConnectionHealth) Connection state changed to Controller::StateConfirming
[07.04.2022 13:38:53.702] Debug: (networking - ConnectionHealth) Starting unsettled period.
[07.04.2022 13:38:53.702] Debug: (networking - ConnectionHealth) ConnectionHealth deactivated
[07.04.2022 13:38:53.702] Debug: (networking - PingHelper) PingHelper deactivated
[07.04.2022 13:39:03.309] Debug: (captiveportal - CaptivePortalDetection) Controller/Stability state changed
[07.04.2022 13:39:03.309] Info: (captiveportal - CaptivePortalDetection) connecting, stopping background captive-portal Monitor
```

as you can see, the pinghelper deactivation takes ~10 seconds.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

I was able to reproduce the hanging/freezing state, on Mozilla VPN 2.8.0 (2.202204062129), on Windows 11, while disconnecting from the internet.

Attaching a video and logs - time: 13:38 - 13:39.

!disconnect from the internet.mp4|width=496,height=748!

[^13-39.txt]

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

I can reproduce with Bianca Hidecuti 's approach of being connected to the VPN and killing the WiFi. Is this really because we are waiting for the ICMP to come back and this freezes the UI?

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

I believe this has been around for a long time now. 😞  We should fix it but it’s not a blocker.

**Comment by data-sync-user** [comment]
➤ Owen Kirby commented:

From the description, this should be fixed by PR #4521 ( https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4521 )

**Comment by data-sync-user** [comment]
➤ Santiago Andrigo commented:

I just re-tested and the app doesn’t hang like it used to, in the scenario of disconnecting the WiFi. Tested on Windows 11, version 2.10.0 (2.202209220108). Sending back to QA.

**Comment by data-sync-user** [comment]
➤ Bianca Hidecuti commented:

Verified this as fixed on Mozilla VPN 2.10.0 (2.202209221847), while using Windows 11 and I am no longer able to reproduce the hanging/freezing state when disconnecting from the internet while the VPN is ON.

Also, if using Clumsy to slow down the connection, the client is not freezing anymore.

Attaching postfix video.

!bandicam 2022-09-23 17-12-06-3831.mp4|width=1648,height=1076!
